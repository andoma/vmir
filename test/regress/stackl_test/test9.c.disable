//#include <sysio.h>

typedef struct
{
    int aa;
    int bb;
    char cc;
} struct_t;

int main()
{
    prints("Testing pointer arithmetic\n");

    {
        char data[10];
        int ii;

        for (ii=0; ii<10; ii++)
        {
            data[ii] = (char)ii;
        }

        char *ptr;
        ptr = data;
        for (ii=0; ii<10; ii++)
        {
            if (*ptr++ != data[ii]) 
            {
                prints("pointer error ");
                printi((int)ptr);
                prints(" ");
                printi((int)&data[ii]);
                prints("\n");
            }
            else
            {
                prints("Index ");
                printi(ii);
                prints(" OK\n");
            }
        }
    }

    {
        char *cptr;
        int  *iptr;
        struct_t *sptr;
        void *vptr;
        int  ii;

        cptr = 0;
        iptr = 0;
        sptr = 0;
        vptr = 0;
        for (ii=0; ii<10; ii++)
        {
            if (cptr++ != (char*)ii) 
                prints("char pointer arithmetic error\n");
            else
                prints("char pointer arithmetic OK\n");

            if (iptr++ != (int *)(ii*4)) 
                prints("int pointer arithmetic error\n");
            else
                prints("int pointer arithmetic OK\n");

            if (sptr++ != (struct_t*)(ii*sizeof(struct_t)) )
                prints("struct pointer arithmetic error\n");
            else
                prints("struct pointer arithmetic OK\n");

            if (vptr++ != (void*)ii) 
                prints("void pointer arithmetic error\n");
            else
                prints("void pointer arithmetic OK\n");
        }
    }

    prints("Test complete\n");
    return 0;
}
