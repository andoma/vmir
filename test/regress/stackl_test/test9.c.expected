Testing pointer arithmetic
Index 0 OK
Index 1 OK
Index 2 OK
Index 3 OK
Index 4 OK
Index 5 OK
Index 6 OK
Index 7 OK
Index 8 OK
Index 9 OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
char pointer arithmetic OK
int pointer arithmetic OK
struct pointer arithmetic OK
void pointer arithmetic OK
Test complete
